plugins/rbac/src/hooks/useSelectedMembers.ts

```diff
@@ -10,23 +10,42 @@ import { SelectedMember } from '../components/CreateRole/types';
 import { MemberEntity } from '../types';
 import { getSelectedMember } from '../utils/rbac-utils';
 
+export const useRole = (
+  roleEntityRef: string,
+): {
+  loading: boolean;
+  role: Role | undefined;
+  roleError: Error;
+} => {
+  const rbacApi = useApi(rbacApiRef);
+  const {
+    value: roles,
+    loading,
+    error: roleError,
+  } = useAsync(async () => await rbacApi.getRole(roleEntityRef));
+
+  return {
+    loading,
+    role: Array.isArray(roles) ? roles[0] : undefined,
+    roleError: (roleError as Error) || {
+      name: (roles as Response)?.status,
+      message: `Error fetching the role. ${(roles as Response)?.statusText}`,
+    },
+  };
+};
```

Comment:
@AndrienkoAleksandr can you move this hook to a separate file?

Reply:
Done.